### ISSUE TITLE
Don't select the whole table

### ISSUE THREAD

**Issue by nitedani** [MichalLytek/typegraphql-prisma#284]
**Is your feature request related to a problem? Please describe.**
The following query pulls the whole posts table in memory:
`
query{
 posts{
  id
 }
}
`
If there are a large number of posts with large content, this can be an issue.

**Describe the solution you'd like**
Transform the selected graphql fields into prisma select and pass it to the prisma client.

**Additional context**
https://github.com/paljs/prisma-tools/blob/main/packages/plugins/src/select.ts

**Comment by MichalLytek** [comment]
This can be only opt-in because you may have field resolvers that depend on some not requested field.

**Comment by tylergets** [comment]
@MichalLytek would you consider an exclude feature or something? I have one table which is large but I only need a small bit of it. Trying to find best way to exclude a certain field from the prisma query

**Comment by MichalLytek** [comment]
You can always eject and use your custom resolver instead of the single generated one.

**Comment by SebastianGrzymala** [comment]
You may use following approach to query only for choosen fields in your custom resolver. It works fine till you work on one table.
```
import graphqlFields from 'graphql-fields';
...

@TypeGraphQL.Resolver(_of => SomeTable)
...
// Build prisma select object based on graphql query fileds
const select = Object.keys(
    graphqlFields(info, {}, { excludedFields: ['__typename'] }),
  ).reduce(
    (acc, key) => ({
      ...acc,
      ...{ [key]: true },
    }),
    {},
  );

return getPrismaFromContext(ctx).sometable.findMany({
    ...args,
    select
});

```

**Comment by chriszrc** [comment]
@MichalLytek This seems like a really important feature, the auto generated resolvers should be populating the select field just with the columns requested...

**Comment by MichalLytek** [comment]
@chriszrc This feature is not that simple. Imagine you have a field resolver that requires some other fields. Then the simple solution of parsing graphql request tree would break it. So the proper resolution would require some decorator/metadata to register dependent fields too.

**Comment by chriszrc** [comment]
@MichalLytek Hmm, I'm not sure I follow, afaict, the only place @FieldResolvers are used in the autogenerated classes are in the *RelationsResolvers, and in order to make a graphql request for related entities, you'd need to include the all the relevant info to request nested relations/columns. Wouldn't your scenario only play out if one were customizing the generated classes to begin with? I think limiting the fields would certainly be a better default, but I guess it could be an option during generation?

**Comment by MichalLytek** [comment]
I'm talking about this scenario:
https://prisma.typegraphql.com/docs/advanced/adding-fields

With the prisma select applied, your field resolver can no longer expect the fields to be present.

**Comment by chriszrc** [comment]
Right, but these aren't part of the default generation, they're customized additions, so why make everyone who's using the graphql api as is suffer
